src/SDK/SmartDetectors.ProjectTemplate/LogSearchSmartDetector.cs

```diff
@@ -30,22 +30,18 @@ public class $detectorName$ : ISmartDetector
         public async Task<List<Alert>> AnalyzeResourcesAsync(AnalysisRequest analysisRequest, ITracer tracer, CancellationToken cancellationToken)
         {
             tracer.TraceInformation("Analyzing the specified resources...");
-            $if$ ("$dataType$" == "Log Analytics")
             // Get the Log Analytics client
-            ITelemetryDataClient dataClient = await analysisRequest.AnalysisServicesFactory.CreateLogAnalyticsTelemetryDataClientAsync(new List<ResourceIdentifier>() { analysisRequest.RequestParameters.TargetResources.First() }, cancellationToken);
-            $else$
-            // Get the Application Insights client
-            ITelemetryDataClient dataClient = await analysisRequest.AnalysisServicesFactory.CreateApplicationInsightsTelemetryDataClientAsync(new List<ResourceIdentifier>() { analysisRequest.RequestParameters.TargetResources.First() }, cancellationToken);
-            $endif$
+            ILogAnalyticsClient dataClient = await analysisRequest.AnalysisServicesFactory.CreateLogAnalyticsClientAsync(analysisRequest.RequestParameters.TargetResources.First(), cancellationToken);
+            
             // Run the query 
-            IList<DataTable> dataTables = await dataClient.RunQueryAsync(@"$tableName$ | count", cancellationToken);
+            IList<DataTable> dataTables = await dataClient.RunQueryAsync(@"$tableName$ | count", TimeSpan.FromMinutes(5), cancellationToken);
```

Comment:
>TimeSpan.FromMinutes(5) [](start = 97, length = 23)

It will be better to define a const and use it in both query executions. Even better if you have configuration

Reply:
Also how did you decide, it should be 5 min? 

---
In reply to: [301935869](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/68#discussion_r301935869) [](ancestors = 301935869)

Reply:
changed to 1 day

---
In reply to: [301938403](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/68#discussion_r301938403) [](ancestors = 301938403,301935869)